src/redis-cli.c

```diff
@@ -2755,12 +2806,32 @@ static int parseOptions(int argc, char **argv) {
             config.bigkeys = 1;
         } else if (!strcmp(argv[i],"--memkeys")) {
             config.memkeys = 1;
-            config.memkeys_samples = 0; /* use redis default */
+            config.memkeys_samples = -1; /* use redis default */
         } else if (!strcmp(argv[i],"--memkeys-samples") && !lastarg) {
             config.memkeys = 1;
             config.memkeys_samples = atoi(argv[++i]);
         } else if (!strcmp(argv[i],"--hotkeys")) {
             config.hotkeys = 1;
+        } else if (!strcmp(argv[i], "--keystats")){
+            config.keystats = 1;
+            config.memkeys_samples = -1; /* use redis default */
+        } else if (!strcmp(argv[i],"--keystats-samples") && !lastarg) {
+            config.keystats = 1;
+            config.memkeys_samples = atoi(argv[++i]);
```

Comment:
i see in the code, we allow n to be negitive (memkeys-samples <n> is the same)?

Reply:
Good point. Even if `--memkeys_samples` could be -1 (not to use `SAMPLES`), the value entered by the user should not be negative.

I changed `unsigned memkeys_samples;` to be an `int` to solve https://github.com/redis/redis/pull/9735.

We could add something like for `--cursor` or `--top`:
```
if (isPositiveInteger(argv[i+1])) {
    config.keystats = 1;
    config.memkeys_samples = atoi(argv[++i]);
} else {
    fprintf(stderr, "--keystats-samples cannot be followed by a negative number.\n");
    exit(1);
}
```

And do the same for `--memkeys-samples`.

Reply:
The fact `--memkeys-samples` is using `atoi()` made me use an `int`.
If we want to be accurate, we should use a `long long`, like it is done in `memoryCommand()`, and check the value to be [0, LLONG_MAX].

Reply:
`memkeys_samples` was changed to `long long` and the argument value in `--memkeys-samples` and `--keystats-samples` is being checked.
We now check for `strtoull()` parsing issues (`--cursor` and `--top`) and did the same for `strtoll()` (`--keystats-samples` and `--memkeys-samples`). I did not check for `ERANGE` as I thought it was overkill.